src/Validation/DNSCheckValidation.php

```diff
@@ -57,21 +89,46 @@ public function getWarnings()
     protected function checkDNS($host)
     {
         $variant = INTL_IDNA_VARIANT_2003;
-        if ( defined('INTL_IDNA_VARIANT_UTS46') ) {
+        if (defined('INTL_IDNA_VARIANT_UTS46')) {
             $variant = INTL_IDNA_VARIANT_UTS46;
         }
         $host = rtrim(idn_to_ascii($host, IDNA_DEFAULT, $variant), '.') . '.';
 
-        $Aresult = true;
-        $MXresult = checkdnsrr($host, 'MX');
+        // Get all MX, A and AAAA DNS records
+        $dnsRecords = dns_get_record($host, DNS_MX + DNS_A + DNS_AAAA);
 
-        if (!$MXresult) {
-            $this->warnings[NoDNSMXRecord::CODE] = new NoDNSMXRecord();
-            $Aresult = checkdnsrr($host, 'A') || checkdnsrr($host, 'AAAA');
-            if (!$Aresult) {
-                $this->error = new NoDNSRecord();
+        // No MX, A or AAAA records
+        if (empty($dnsRecords)) {
+            $this->error = new NoDNSRecord();
+            return false;
+        }
+
+        $aRecords  = [];
+        $mxRecords = [];
+
+        // Iterate over all returned DNS records
+        foreach ($dnsRecords as $dnsRecord) {
+            if ($dnsRecord['type'] == 'MX') {
+                // "Null MX" record indicates the domain accepts no mail (https://tools.ietf.org/html/rfc7505)
+                if (empty($dnsRecord['target']) || $dnsRecord['target'] == '.') {
+                    $this->error = new DomainAcceptsNoMail();
+                    return false;
+                }
+
+                $mxRecords[] = $dnsRecord;
+                continue;
+            }
+
+            if (in_array($dnsRecord['type'], ['A', 'AAAA'])) {
+                $aRecords[] = $dnsRecord;
```

Comment:
Whiy is ```$aRecords``` needed? I have failed to see usage of this var.

Reply:
Not currently used. I was thinking about pinging or validating the target address(es). Technically, it could currently still be invalid/local. Probably getting towards overkill there though... I'll remove it.